Tesla and supermatter zaps can trigger electrolysis reactions. Hypernoblium to antinoblium conversion requires supermatter zaps.

Pickle-Coding opened the pull request:
## About The Pull Request
Supermatter and tesla zaps can trigger electrolysis reactions. Hypernoblium to antinoblium conversion requires supermatter zaps above 5 GeV, with 100% efficiency at 9GeV. Antinoblium is generated at 1:1 from hypernoblium consumed. The requirement for hypernoblium conversion is also included in the reaction console.
## Why It's Good For The Game
Ordinary electrolysis is a rather boring way to obtain antinoblium, especially with fire extinguisher cheese. This limits the creation of antinoblium to more extreme and loud methods. This is mainly intended for balancing #91016, but seems out of scope to include in that PR. This also gives room for antinoblium to be balanced as one of the more exotic gases, and is much more obvious when antinoblium is being created.
## Changelog
:cl:
add: Supermatter and tesla zaps can trigger electrolyzer reactions.
balance: Hypernoblium to antinoblium electrolysis requires supermatter zaps above 5 GeV.
balance: Hypernoblium is converted to antinoblium 1:1 during electrolysis.

/:cl:

Pickle-Coding committed:
Tesla and supermatter electrolysis.

Tesla and supermatter zaps will trigger electrolysis reactions.

Hypernoblium electrolysis requires supermatter zap power_level to be passed in, and creates antinoblium at 1:1 of hypernoblium consumed.

Deadgebert commented:
Ah, so the previous PR turned out to be too griefy so now we're making atmos more difficult so only you and a few others can do it?

Pickle-Coding commented:
> Ah, so the previous PR turned out to be too griefy so now we're making the production more difficult so only you and a few others can do it?

It's not to limit knowledge to a few individuals, it's to prevent being able to make it in a stealthy and effortless way. You can look at the gas monitor consoles to find out the requirements of the reaction.

> The benefits of antinoblium are zaukerite, supermatter cascade, and as HFR fuel. None of this is common enough to warrant making it more difficult to achieve.

Zaukerite is cosmetic and is meant to show your atmos prowess. Supermatter cascade ends the entire round. Antinoblium and hypernoblium HFR fuel is the most dangerous and highest tier HFR fuel, which can duplicate gases due to its extreme efficiency and also almost guaranteed to end the round if it explodes.

LemonInTheDark commented:
> Ah, so the previous PR turned out to be too griefy so now we're making the production more difficult so only you and a few others can do it?
> 
> The benefits of antinoblium are zaukerite, supermatter cascade, and as HFR fuel. None of them are common enough to warrant making them more difficult to achieve.

I need you to be like 80% less hostile or I'm gonna throw you into the ocean

Deadgebert commented:
> > Ah, so the previous PR turned out to be too griefy so now we're making the production more difficult so only you and a few others can do it?
> 
> I need you to be like 80% less hostile or I'm gonna throw you into the ocean

you're right, sorry, i've edited it

AtomTheProphet commented:
It would be cool (and possibly allow for future expansion in how reactions can take place in atmospheres in general, probably) for supermatter and tesla zaps to be able to trigger electrolysis in gases [Supermatter shard powered electrolysis tank?], but it should not be something that certain reactions are limited to, in my opinion.

sippykot commented:
> Tesla and supermatter zaps can trigger electrolysis reactions.

Good change!

> Hypernoblium to antinoblium conversion requires supermatter zaps.

certified /tg/ 🪣

LemonInTheDark commented on code/__DEFINES/reactions.dm:
```diff
@@ -261,3 +261,8 @@
 #define PN_BZASE_NUCLEAR_PARTICLE_RADIATION_ENERGY_CONVERSION 2.5
+
+// Electrolysis:
+// Electrolysis arguments:
+/// Supermatter power argument.
```
what?

LemonInTheDark commented on code/modules/power/supermatter/supermatter.dm:
```diff
@@ -1050,8 +1050,19 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
-		pressure = max(1,target_turf.return_air().return_pressure())
+		var/datum/gas_mixture/air_mixture = target_turf.return_air()
+		pressure = max(1, air_mixture.return_pressure())
+		for(var/reaction in GLOB.electrolyzer_reactions)
+			var/datum/electrolyzer_reaction/current_reaction = GLOB.electrolyzer_reactions[reaction]
+
+			if(!current_reaction.reaction_check(air_mixture = air_mixture, electrolyzer_args = list(ELECTROLYSIS_ARGUMENT_SUPERMATTER_POWER = power_level)))
+				continue
+
+			current_reaction.react(location = target_turf, air_mixture = air_mixture, working_power = zap_str / 200, electrolyzer_args = list(ELECTROLYSIS_ARGUMENT_SUPERMATTER_POWER = power_level)) // Only a portion of the zap_str contributes to working power.
```
seems like something we should have a wrapper for

Pickle-Coding replied:
Yeah, or actually a gas mixture proc.

Sakamoto4ka commented:
Sounds like headache and 1 week without shower

moocowswag commented:
How does this work though, does the zap have to go through the tile? does it have to hit the container/pipe if its in one?

Kiromaka commented:
Same as in your other PR - dont change a gas that already has uses and interactions into something completely different, especially dont make it 20x harder to get because of the change.

Kiromaka commented:
> especially with fire extinguisher cheese

Also - the hell does this have to do with anti-nob production, fire extinguisher is used for nob production.

Pickle-Coding commented:
> Same as in your other PR - dont change a gas that already has uses and interactions into something completely different, especially dont make it 20x harder to get because of the change. Make a new gas, but the idea of a gas that is just a money printer and a 1 click shuttle call is just trash.

We have plenty of gases that do nothing. Better to make them do things than add another gas. The methods of obtaining one of the more niche gases isn't sacred. Argue for why this requirement for creating the gas is bad instead of treating the current method as intrinsically good.

> > especially with fire extinguisher cheese
> 
> Also - the hell does this have to do with anti-nob production, fire extinguisher is used for nob production.
> 
> Its easier to cool a heated anti-nob chamber with intended tools than to set it on fire.

Plasma, oxygen and nitrogen mixture with super saturation thresholds by an electrolyzer. Burn that to create tritium, then extinguish it for hypernoblium, which will then create antinoblium as it's also cold.

Pickle-Coding commented:
> How does this work though, does the zap have to go through the tile the gasses are? does it have to hit the container/pipe if they are in one?
> 
> You also including tesla in the title, will tesla coils with enough power be able to do it?

Electrolysis will occur where the zap hits. Zaps from any source can trigger normal electrolysis reactions.

Pickle-Coding committed:
Add proc for electrolyzing a gas mixture.

Adds the /datum/gas_mixture/proc/electrolyze() proc. Calls for every electrolyzer_reaction reaction_check() then react() when the checks don't fail. Replaces manual instances of doing this with this proc. Also removes unused location parameter for electrolyzer_reaction.react().

LemonInTheDark commented:
democracy was never alive. player input is relevant and I take it into consideration but it can't be the only deciding factor.
if you wanna chat about why I merged it feel free to bug me I LOVE to yap.

sippykot commented:
why it's good for the game? "its intended to enable another dogshit PR that everyone hates"

Kiromaka commented:
> democracy was never alive. player input is relevant and I take it into consideration but it can't be the only deciding factor. if you wanna chat about why I merged it feel free to bug me I LOVE to yap.

Didn't we go over this with wallening already. Player input is the most important factor. And I could see your argument if it was like 30/70 split. But this is literally a unanimously hated PR, that enables another unanimously hated PR. This PR also makes no sense outside of being merged with the other PR.  Balance changes are also under more scrutiny in terms of player and admin input. 

Anti-nob now is almost fully an antag gas, cause to create it you need to release a tesla or edge the super matter extremly close to going tesla, how is this exactly good for the game ?

LemonInTheDark commented:
> why it's good for the game? "its intended to enable another dogshit PR that everyone hates"

I like the idea of making antinob hard to get because it is constructed around this idea of the "atmos wizard" and that dream gains weight if more well, weight is put behind it.

Kiromaka commented:
> I do not understand why it would remove the cascade fully, could you explain? 

I might be blind - but I don't see an exception for anit-nob converting nob into anti-nob with its replication reaction in that PR.

That means, that it will remove cascade from the game. Just a reminder - protot is gone, no more funny crystal. 

If exception is added (or there is and i'm blind) - anti-nob will consume SM waste and mediums like N2, braking the 40/40 balance and requiring some goofy ahh setups for Cascade, which is already extremly easy to prevent before timer - removing it from the game in a practical sense, unless everyone just ignores the giant (attention crew, we are fucked) message. The goofy ahh setup will also slow down the cascade reaction - fast decay = waste, waste = more anti-nob, more anti-nob = more nob and nob isnt easy to produce in large quantities.

LemonInTheDark commented:
> I might be blind - but I don't see an exception for anit-nob converting nob into anti-nob with its replication reaction in that PR.

reactions have a sorting system, we order them into buckets which can occur at any time between them, and then check all their conditions before reacting. All except for hypernob, who has a hardcoded check for his conditions directly written into the react() proc, before anything else (it's faster this way). No exception is needed because any significant amount of hypernob would block replication from occurring.

> If exception is added (or there is and i'm blind) - anti-nob will consume SM waste and mediums like N2, braking the 40/40 balance and requiring some goofy ahh setups for Cascade, which is already extremly easy to prevent before timer - removing it from the game in a practical sense, unless everyone just ignores the giant (attention crew, we are fucked) message. The goofy ahh setup will also slow down the cascade reaction - fast decay = waste, waste = more anti-nob, more anti-nob = more nob and nob isnt easy to produce in large quantities.

Similarly not a concern because of the above. I understand why you would be worried but you did not have all the relevant information.

Unit2E commented:
For what it's worth: Long time atmos main, and I do like this PR.

It's a much more interesting idea than how it currently works when looking at just antinoblium, and is otherwise a fun mechanic as well. Personally, I think there are a handful of gases that are either very uninteresting to make or have lacking uses. Antinoblium is primarily used for Cascades, bombs (though never necessary), bragging, and arguably Zaukerite bolts. But it's also just both really boring and very easy to make (odd for a gas that has bragging as a main use)

Doing it in the HFR is more interesting than electrolyzing was, but not difficult overall (just not known about). That whole thing might need changing, it's not exceedingly difficult to get 10000s of moles of the stuff just running the HFR. The idea that this new method is going to seriously interfere with Cascades is silly. Cascades SHOULD be very difficult in the first place. They're currently not too much, really. 

Protecting the sanctity of an obscure SM delam that straight up ends the round for all players is a strange viewpoint to have in my mind. I'd consider myself very proficient at atmos, and I don't believe at all I should be able to force a round end within 30 minutes just because I did atmos for a while and built reinforced walls. All changes to gases that have fallen by the wayside, and generally make the game more interesting, should be welcomed, and that includes the other PR.

Kiromaka commented:
> and I don't believe at all I should be able to force a round end within 30 minutes just because I did atmos for a while and built reinforced walls. All changes to gases that have fallen by the wayside, and generally make the game more interesting, should be welcomed, and that includes the other PR.

Cascade is very easy to prevent, it announces itself 10 minutes before the delam. Seeing a N2 filling station + bz + trit, all being done at the same time should at least raise an eyebrow, especially with no HFR present. Any competent cap, or CE (if CE isnt the one doing it) will be able to prevent cascade. I cant remember a single cascade (excluding the ones being done in hopes of after-timer prevention) on live servers. 

Making a cascade setup isnt hard ye. Cascading on live - undoable with competent crew. Unless you build in like 5 layers of reinforced walls around SM - but you not doing that on more than 40 pop.

Tostah commented:
My god. I can’t believe the things I’m reading… 

“player input shouldn’t decide weather something gets merged or not” 

?????????????????????????????????

“Maintainers exist to make decisions for players” 

bro. Fucking hell. I really feel like there isn’t a need to explain this, because it’s self evident! Yes, maintainers need to make changes without input from players. Nerfs, buffs, changes in general. Maintainers do this things automatically because we put trust in them that they’ll do what’s best for the playerbase. 

But if the playerbase nearly collectively decides ‘yeah this is fucking stupid’ then maintainers need to listen to that. It shouldn’t need to come down to a server boycot.

Why? 

Because it’s the players who are playing the game. They deal with this new mechanic and all its ins and outs. This simple fact makes players the most important resource for any game developer. 

This is game dev 101. Games have existed for decades, and people have gotten good at making them. If there’s ONE LESSON that video game developers have learned over the years, it’s;

LISTEN TO YOUR FUCKING PLAYERBASE. 

If you don’t, you will alienate them. The game will die. And that will be that.

LemonInTheDark commented:
That very explicitly is not the things I said, I chose my wording very carefully and I'd appreciate it if you didn't crash out on the things you think I said.

I am not going to kill a pr I think will be cool and engaging without trying it because it got a lot of people upset for reasons I do not find compelling. This is not the same as ignoring the playerbase (such as it is). I understand why you guys are worried, and I don't think it'll go down like that. I am willing to be proven wrong in that respect.

I think that covers most of what you've posted, to target the things I think it does not:

> LISTEN TO YOUR FUCKING PLAYERBASE.
>
>If you don’t, you will alienate them. The game will die. And that will be that.
>
>This is even more true for TG. There are severs which are very similar to TG. There are reasonable alternatives. You forget that we are convincing people to play on this server instead of all the other ones.

I understand if you are anxious about population counts, they are important to be aware of. I do not agree that they should be used as a direct metric (we are trying to build the server/codebase "we" (the community/maintainers/admins) want, not the one that pulls the most players, ideally both happen but if the focus is on the first then we have killed the larger community), but in this case I don't think it's material because I don't think these prs will in the end alienate people.

> Your opinion as a maintainer is not better than everyone else’s just because you can write code.

It is a fact of our repository that maintainers get to make decisions about what prs are merged and closed. That authority comes from a decision by the rest of the team based off someone's understanding of the thing we are all collectively working on, and "vibing correctly". Writing good code is a requirement only because that is the medium in which we work, and if you can't speak that language you'll merge bullshit.
